### DISCUSSION TITLE
对于包体积的一点疑问

### DISCUSSION THREAD

**Discussion by flx413** [alibaba/xquic#17] [General]
我想编译android版本的xquic，用的是xqc_build.sh脚本编译，但我发现编译出来包体积有些大。我的编译环境是：

机器：Ubuntu 18.04 64位
工具链：android ndkr16b、clang
使用的是boringssl

我的操作：
```
sh xqc_build.sh android android_build android_release /home/name/Code/xquic/third_party/boringssl

// 等待编译

virtual-machine:~/Code/xquic/android_release/arm64-v8a$ ls -l
总用量 38772
-rw-rw-r-- 1 gg gg  6438476 1月   7 20:24 libboringssl_gtest.a
-rw-rw-r-- 1 gg gg 12056486 1月   7 20:24 libcrypto.a
-rw-rw-r-- 1 gg gg   438584 1月   7 20:24 libdecrepit.a
-rw-rw-r-- 1 gg gg 10016270 1月   7 20:24 libssl.a
-rw-rw-r-- 1 gg gg  1882264 1月   7 20:24 libtest_support_lib.a
-rwxrwxr-x 1 gg gg  8856672 1月   7 20:24 libxquic.so
virtual-machine:~/Code/xquic/android_release/arm64-v8a$

```
脚本没有改动，我编译出来有8M，但是淘宝apk里的包只有1M左右

**Comment by Kulsk** [comment]
it's not stripped, try strip command

**Reply by flx413** [reply]
thanks, it works

**Reply by robinhzp** [reply]
cmake -DSSL_TYPE=${SSL_TYPE_STR} -DSSL_PATH=${SSL_PATH_STR} -DSSL_INC_PATH=${SSL_INC_PATH_STR} -DSSL_LIB_PATH=${SSL_LIB_PATH_STR} -DCMAKE_BUILD_TYPE=MinSizeRel  ..

For Linux, the above command will automatically strip symbols.

**Reply by yangqingyuan** [reply]
@robinhzp 
我看xqc_build.sh 里面已经有对应的配置了，为啥还是8m多
机器：mac 64位
工具链：android ndkr20b、clang
使用的是boringssl

![image](https://user-images.githubusercontent.com/6867757/150736458-a50b2fb7-20f9-4dae-9607-edcaf36bade5.png)

**Comment by robinhzp** [comment]
当前cmake文件是把ssl和crypto库都静态打包在libxquic里面的，还有符号表没有strip，所以比较大


发自我的iPhone

------------------ 原始邮件 ------------------
发件人: yangqingyuan ***@***.***&gt;
发送时间: 2022年1月24日 15:02
收件人: alibaba/xquic ***@***.***&gt;
抄送: robinhzp ***@***.***&gt;, Mention ***@***.***&gt;
主题: 回复：[alibaba/xquic] 对于包体积的一点疑问 (Discussion #17)

**Comment by ag2s20150909** [comment]
```shell
"$ANDROID_NDK"/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android-strip -s libxquic.so -o libxquic.so.strip
```
